src/snowcli/utils.py

```diff
@@ -89,62 +103,72 @@ def parse_requirements(requirements_file: str = "requirements.txt") -> list[str]
-                snowflake_packages.append(
-                    f"{package}",
-                )
+            if package.name.lower() in channel_data["packages"]:
+                snowflake_packages.append(package)
             else:
                 click.echo(
-                    f'"{package}" not found in Snowflake anaconda channel...',
+                    f'"{package.name}" not found in Snowflake anaconda channel...',
                 )
-                other_packages.append(package)
-        # As at April 2023, streamlit appears unavailable in the Snowflake Anaconda channel
-        # but actually works if specified in the environment
-        if "streamlit" in other_packages:
-            other_packages.remove("streamlit")
-        return {"snowflake": snowflake_packages, "other": other_packages}
+                if package.name.lower() == "streamlit":
+                    # As at April 2023, streamlit appears unavailable in the Snowflake Anaconda channel
+                    # but actually works if specified in the environment
+                    click.echo(
+                        f'"{package.name}" is not available in the Snowflake anaconda channel but is supported in the environment.yml file',
+                    )
+                else:
+                    other_packages.append(package)
+        return SplitRequirements(snowflake=snowflake_packages, other=other_packages)
     else:
-        click.echo(f"Error: {response.status_code}")
-        return {}
+        click.echo(f"Error reading Anaconda channel data: {response.status_code}")
+        raise typer.Abort()
 
 
 def generate_streamlit_environment_file(
     excluded_anaconda_deps: Optional[List[str]],
+    requirements_file: str = "requirements.snowflake.txt",
 ) -> Optional[Path]:
     """Creates an environment.yml file for streamlit deployment, if a Snowflake
     requirements file exists.
     The file path is returned if it was generated, otherwise None is returned.
     """
-    if os.path.exists("requirements.snowflake.txt"):
-        # for each line in requirements.snowflake.txt, prepend '- ' to the line and prepare it for interpolation into the template
-        with open("requirements.snowflake.txt", "r", encoding="utf-8") as f:
-            requirements = f.read().split("\n")
+    if os.path.exists(requirements_file):
+        snowflake_requirements = parse_requirements(requirements_file)
+
         # remove explicitly excluded anaconda dependencies
         if excluded_anaconda_deps is not None:
             print(f"""Excluded dependencies: {','.join(excluded_anaconda_deps)}""")
-            requirements = [
-                line for line in requirements if line not in excluded_anaconda_deps
+            snowflake_requirements = [
+                r
+                for r in snowflake_requirements
+                if r.name not in excluded_anaconda_deps
             ]
         # remove duplicates, remove comments, remove snowflake-connector-python
```

Comment:
Is this comment still true? Do we remove duplicates? Can you add tests for this?

Reply:
we do via the `list(set())`, but I should probably move that into parse_requirements so that callers don't need to.
I'll do that now and write a test to cover it.

Reply:
created `utils.deduplicate_and_sort_reqs` with test case